### PR TITLE
Tag Tracking+: Sidebar filtering option; no-tags placeholder

### LINKED ISSUE #832: Tag Tracking+: Improve sidebar UI with no tracked tags
### Concept

If your account follows no tags, the Tag Tracking+ sidebar looks a bit weird. A placeholder might make sense here.

(This could maybe be an opportunity to educate users on how you actually follow a tag, though there's not much room and adding a button that pops up an explainer modal would be quite extreme. The thought did cross my mind though.)

### LINKED ISSUE #833: Tag Tracking+ collapse 0 tags option
### Discussed in https://github.com/AprilSylph/XKit-Rewritten/discussions/830

<div type='discussions-op-text'>

<sup>Originally posted by **nobodysuspectsthebutterfly** November 17, 2022</sup>
Thanks so much for adding the sidebar option to Tag Tracking+!  Much appreciated. ❤

But a tiny feature request -- New Xkit's TT+ had an option for "only show tags with new posts", so that when a tag was at 0, it wouldn't show in the list of tags on the sidebar. It was always a fave of mine. I'd love to see it as an option here if you get a chance. Thanks!</div>

### PR DESCRIPTION
**Pull request by marcustyphoon**
### Description

This adds an option to only show tags with new posts in the sidebar, and adds status-indicating placeholder elements to the sidebar for a) not having any tracked tags, b) having tracked tags, all of which are hidden by the option, and c) having tracked tags which are still loading their counts and are hidden by the option.

I think they should probably look different than regular sidebar element list items, but I have no ideas for the CSS. Head empty.

This code is kind of messy since we're manipulating the utility-generated sidebar DOM elements. I keep edging into model-view-controller type structures ("make it look like react") and backing off.

resolves #832, resolves #833

### Testing steps

- Enable the sidebar.
- With the new "only show tags with new posts in the sidebar" option disabled, follow 0 tags. There should be a status placeholder in the sidebar.
- Follow some tags and refresh. There should be no status placeholder, all of the tags should appear, and they should update their counts in sequence. Opening the search dropdown mid-update should cause no problems.
- With the new "only show tags with new posts in the sidebar" option enabled, track some tags with new posts (or add `if (Math.random() < 0.3) unreadCount = 5;` at line 56.) There should be a loading placeholder in the sidebar, then some tags with counts. Opening the search dropdown mid-update should cause no problems.
- Clear the new posts. There should now be a status placeholder in the sidebar. Hard refresh. There should be a loading placeholder in the sidebar, then a a status placeholder.
- Open a page with no sidebar (https://www.tumblr.com/april) and soft navigate to the dashboard; no problems should occur.

### CODE REVIEW

File: src/scripts/tag_tracking_plus.css [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#tag-tracking-plus ul::after {
+  display: flex;
+  justify-content: space-between;
```
Comment (AprilSylph) [inline src/scripts/tag_tracking_plus.css:3]:
```suggestion
  justify-content: center;
```

File: src/scripts/tag_tracking_plus.css [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  align-items: center;
+  min-height: 36px;
+  padding: 5px 10px;
+  color: rgba(var(--white-on-dark), 0.6);
```
Comment (AprilSylph) [inline src/scripts/tag_tracking_plus.css:7]:
```suggestion
  padding: 5px 10px;
  border-radius: 3px;
  margin-top: 10px;

  background-color: rgba(var(--white-on-dark), 0.13);
  color: rgba(var(--white-on-dark), 0.6);
  font-size: .875rem;
```

File: src/scripts/tag_tracking_plus.css [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+#tag-tracking-plus ul:empty::after {
+  content: 'No tracked tags';
+}
+
+#tag-tracking-plus.only-show-new[data-loading="true"] ul:not(:empty)::after {
+  content: 'loading...';
```
Comment (AprilSylph) [inline src/scripts/tag_tracking_plus.css:15]:
```suggestion
  content: 'Loading...';
```

File: src/scripts/tag_tracking_plus.css [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+#tag-tracking-plus.only-show-new[data-loading="false"][data-has-new="false"] ul:not(:empty)::after {
+  content: 'No unread tags';
```
Comment (AprilSylph) [inline src/scripts/tag_tracking_plus.css:19]:
```suggestion
  content: 'No unread tags.';
```

File: src/scripts/tag_tracking_plus.css [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  min-height: 36px;
+  padding: 5px 10px;
+  color: rgba(var(--white-on-dark), 0.6);
+}
+
+#tag-tracking-plus ul:empty::after {
+  content: 'No tracked tags';
```
Comment (AprilSylph) [inline src/scripts/tag_tracking_plus.css:11]:
```suggestion
  content: 'No tracked tags!';
```

**Review by AprilSylph — COMMENTED** [review]
Consider:

True Blue | Dark Mode
-------------|----------------
image | image

File: src/scripts/tag_tracking_plus.css [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  align-items: center;
+  min-height: 36px;
+  padding: 5px 10px;
+  border-radius: 3px;
+  margin-top: 10px;
+
+  background-color: rgba(var(--white-on-dark), 0.13);
+  color: rgba(var(--white-on-dark), 0.6);
+  font-size: .875rem;
+}
```
Comment (marcustyphoon) [inline src/scripts/tag_tracking_plus.css:13]: If we felt like it, we could bikeshed this, but I really don't think it's at all important :D

```suggestion
#tag-tracking-plus ul::after {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 2rem;
  padding: 5px 10px;
  border-radius: 3px;
  margin-top: 10px 5px;

  background-color: rgba(var(--white-on-dark), 0.13);
  color: rgba(var(--white-on-dark), 0.6);
  font-size: 1rem;
}
```

Comment (marcustyphoon) [reply src/scripts/tag_tracking_plus.css:13]: additional option:
```css
border: 1px dashed rgba(var(--white-on-dark), 0.13);
```

**Review by AprilSylph — APPROVED** [review]

**Pull request merged**